seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java

```diff
@@ -35,6 +38,8 @@ public final class SeaTunnelRow implements Serializable {
 
     private volatile int size;
 
+    private String partitionName;
```

Comment:
why not use https://github.com/apache/seatunnel/blob/15431c94d9b9f34839173d67c1b638f058f58722/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/CatalogTable.java#L41 ?

Reply:
We read data by partition, then need to write it into same partition too. I guess we have to label the data with partition Name which should be put into SeatunnelRow.

PartitionKey is a different situation, it's automatically created partition based in column value. While partitionName in SeatunnelRow is used for manually created partition.  See https://milvus.io/docs/manage-partitions.md

Pinecone has similar thing called namespace.